components/eamxx/src/share/field/field.hpp

```diff
@@ -206,11 +207,11 @@ class Field {
   //       casting the values to whatever the data type of this field is.
   //       E.g., if data_type()=IntType, you can't pass double's.
   template<HostOrDevice HD = Device, typename ST = void>
-  void update (const Field& x, const ST alpha, const ST beta);
+  void update (const Field& x, const ST alpha, const ST beta, const float fill_val = constants::DEFAULT_FILL_VALUE);
```

Comment:
Can we avoid the hard-coding of float? We spent some time trying to make the functions work with different scalar types (with an eye to something like AD types in the future). It would be nice to keep the generality, and use `ST` as the type for the fill value...